channel/channel_test.go

```diff
@@ -29,12 +31,12 @@ func TestChannel(t *testing.T) {
 
 	testClone := func(t *testing.T) {
 		r := c.Clone()
-		if diff := cmp.Diff(*r, *c, cmp.Comparer(types.Equal)); diff != "" {
```

Comment:
I ran a manual mutation test on `Channel.Clone()` -- I broke it on purpose 241e081695590d468d59afa02efbd97bcd95e92a. Strangely, the code on master successfully detected the problem, whereas `deep.Equal` doesn't seem to?

On this branch / the commit above, if I run `go  test ./channel` 
```
ok  	github.com/statechannels/go-nitro/channel	(cached)
```

```
--- FAIL: TestChannel (0.00s)
    --- FAIL: TestChannel/TestClone (0.00s)
        channel_test.go:33: Clone: mismatch (-want +got):
              channel.Channel(
            - 	{
            - 		Id:             s"0x252e8d12b641d8dd3cde74d299047963c99ce52bded2fbd38afa8e07b76a9306",
            - 		OnChainFunding: s"{[0x0000000000000000000000000000000000000000,0]}",
            - 		FixedPart: state.FixedPart{
            - 			ChainId:           s"9001",
            - 			Participants:      []common.Address{s"0xF5A1BB5607C9D079E46d1B3Dc33f25"..., s"0x760bf27cd45036a6C486802D30B5D9"...},
            - 			ChannelNonce:      s"37140676580",
            - 			AppDefinition:     s"0x5e29E5Ab8EF33F050c7cc10B5a0456D975C5F88d",
            - 			ChallengeDuration: s"60",
            - 		},
            - 		latestSupportedStateTurnNum: 999999999,
            - 		SignedStateForTurnNum: map[uint64]state.SignedState{
            - 			0: {state: state.State{...}, sigs: map[uint]crypto.Signature{}},
            - 			1: {state: state.State{...}, sigs: map[uint]crypto.Signature{}},
            - 		},
            - 	},
            + 	{
            + 		Id:             s"0x252e8d12b641d8dd3cde74d299047963c99ce52bded2fbd38afa8e07b76a9306",
            + 		OnChainFunding: s"{[0x0000000000000000000000000000000000000000,0]}",
            + 		FixedPart: state.FixedPart{
            + 			ChainId:           s"9001",
            + 			Participants:      []common.Address{s"0xF5A1BB5607C9D079E46d1B3Dc33f25"..., s"0x760bf27cd45036a6C486802D30B5D9"...},
            + 			ChannelNonce:      s"37140676580",
            + 			AppDefinition:     s"0x5e29E5Ab8EF33F050c7cc10B5a0456D975C5F88d",
            + 			ChallengeDuration: s"60",
            + 		},
            + 		latestSupportedStateTurnNum: 18446744073709551615,
            + 		SignedStateForTurnNum: map[uint64]state.SignedState{
            + 			0: {state: state.State{...}, sigs: map[uint]crypto.Signature{}},
            + 			1: {state: state.State{...}, sigs: map[uint]crypto.Signature{}},
            + 		},
            + 	},
              )

```

Reply:
I still think it makes sense to go with `go-cmp` as it seems like the more mature library.